db/migrate/20160712044409_create_join_table_storage_profile_storage.rb

```diff
@@ -0,0 +1,10 @@
+class CreateJoinTableStorageProfileStorage < ActiveRecord::Migration[5.0]
+  def change
+    create_table :storage_profile_storages do |t|
+      t.bigint  :storage_profile_id
+      t.bigint  :storage_id
+      t.index [:storage_id, :storage_profile_id], :name => 'storage_to_profile'
+      t.index [:storage_profile_id, :storage_id], :name => 'profile_to_storage'
```

Comment:
I could be wrong on the indexes thing...probably best to run each side of the querying though EXPLAIN and figure out what will or won't be used.  Even so, if *do* you need both directions, prefer:

```ruby
t.index [:storage_id, :storage_profile_id]
t.index :storage_profile_id
```